### PR TITLE
Imported browser shim and samples

### PR DESCRIPTION
**Pull request by justinboswell**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lib/platform.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+export function is_nodejs() {
```
Comment (ColdenCullen) [inline lib/platform.ts]: I'm pretty sure there's an NPM package for this <sup><sub>/s</sub></sup>

File: package.json [hunk 2]
```diff
@@ -5,10 +5,11 @@
   "homepage": "https://github.com/awslabs/aws-crt-nodejs",
   "repository": "github:awslabs/aws-crt-nodejs",
   "contributors": [
-    "Colden Cullen <colden@amazon.com>"
+    "AWS SDK Common Runtime Team <aws-sdk-common-runtime@amazon.com>"
```
Comment (ColdenCullen) [inline package.json]: I'm 90% sure this alias won't accept emails from external addresses, so maybe it doesn't go here

File: package.json [hunk 3]
```diff
@@ -33,18 +35,24 @@
...
     "node-gyp": "^3.8.0",
-    "typescript": "^3.2.2",
+    "typescript": "^3.5.3",
     "yargs": "^13.2.4"
   },
   "dependencies": {
-    "node-pre-gyp": "^0.12.0"
+    "async-mqtt": "^2.3.0",
+    "crypto-js": "^3.1.9-1",
+    "mqtt": "^2.3.1",
```
Comment (ColdenCullen) [inline package.json]: reeeeeeeally wish there was a way to do browser-only dependencies

File: tsconfig.base.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
+{
+    "compilerOptions": {
+        /* Basic Options */
```
Comment (ColdenCullen) [inline tsconfig.base.json]: github no likey

File: lib/browser/mqtt.ts [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+
+    private on_offline = () => {
+        if (this.on_connection_interrupted) {
+            this.on_connection_interrupted(-1);
+        }
+    }
+
+    private on_disconnected = () => {
+        if (this.on_connection_interrupted) {
+            this.on_connection_interrupted(0);
```
[the remark this answers is not in the document]
Comment (justinboswell) [reply lib/browser/mqtt.ts]: Yes. ```WE_WERE_DISCONNECTED_BUT_WE_MEANT_TO_SO_EVERYTHING_IS_STILL_FINE```

File: lib/platform.ts (hunk 1 above)
Comment (justinboswell) [reply lib/platform.ts]: THANKS I HATE IT

Yeah, I know there is, I didn't want to take a dependency for this little bit of code.

File: package.json (hunk 2 above)
Comment (justinboswell) [reply package.json]: We use it on all of our other packages :)

File: package.json (hunk 3 above)
Comment (justinboswell) [reply package.json]: ME TOO. I TRIED!

File: tsconfig.base.json (hunk 4 above)
Comment (justinboswell) [reply tsconfig.base.json]: That's literally from tsc --init

File: lib/browser/crypto.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+export const dummy : boolean = false;
```
[the remark this answers is not in the document]
Comment (justinboswell) [reply lib/browser/crypto.ts]: I'll get it next commit, I have the impl sitting here.

File: lib/browser.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+/* common libs */
+import * as platform from './platform';
+import * as resource_safety from './resource_safety';
+
+/* platform specific libs */
+import * as io from './browser/io';
+import * as mqtt from './browser/mqtt';
+import * as crypto from './browser/crypto';
+
+export {
```
[the remark this answers is not in the document]
Comment (justinboswell) [reply lib/browser.ts]: That syntax doesn't actually work! I was pretty sure I'd tried it before, and sure enough... nope. It SHOULD work. It doesn't.

File: package.json (hunk 2 above)
Comment (ColdenCullen) [reply package.json]: SHIP IT

File: tsconfig.base.json (hunk 4 above)
Comment (ColdenCullen) [reply tsconfig.base.json]: not arguing, just sad

**Pull request merged**